h/streamer/filter.py

```diff
@@ -45,157 +43,51 @@ class FilterHandler(object):
-        'match_of': 'match_of',
-        'lene': 'lene',
-        'leng': 'leng',
-        'lenge': 'lenge',
-        'lenl': 'lenl',
-        'lenle': 'lenle',
-    }
-
     def evaluate_clause(self, clause, target):
-        if isinstance(clause['field'], list):
-            for field in clause['field']:
-                copied = copy.deepcopy(clause)
-                copied['field'] = field
-                result = self.evaluate_clause(copied, target)
-                if result:
-                    return True
+        field_value = resolve_pointer(target, clause['field'], None)
+        if field_value is None:
             return False
+
+        cval = clause['value']
+        fval = field_value
+
+        if isinstance(cval, list):
+            tval = []
+            for cv in cval:
+                tval.append(uni_fold(cv))
+            cval = tval
         else:
-            field_value = resolve_pointer(target, clause['field'], None)
-            if field_value is None:
-                return False
-
-            cval = clause['value']
-            fval = field_value
-
-            if isinstance(cval, list):
-                tval = []
-                for cv in cval:
-                    tval.append(uni_fold(cv))
-                cval = tval
-            else:
-                cval = uni_fold(cval)
+            cval = uni_fold(cval)
 
+        if isinstance(fval, list):
+            tval = []
+            for fv in fval:
+                tval.append(uni_fold(fv))
+            fval = tval
+        else:
+            fval = uni_fold(fval)
+
+        if clause['operator'] == 'one_of':
             if isinstance(fval, list):
-                tval = []
-                for fv in fval:
-                    tval.append(uni_fold(fv))
-                fval = tval
-            else:
-                fval = uni_fold(fval)
-
-            reversed_order = False
-            # Determining operator order
-            # Normal order: field_value, clause['value']
-            # i.e. condition created > 2000.01.01
-            # Here clause['value'] = '2001.01.01'.
-            # The field_value is target['created']
-            # So the natural order is: ge(field_value, clause['value']
-
-            # But!
-            # Reversed operator order for contains (b in a)
-            if isinstance(cval, list) or isinstance(fval, list):
-                if clause['operator'] in ['one_of', 'matches']:
-                    reversed_order = True
-                    # But not in every case. (i.e. tags matches 'b')
-                    # Here field_value is a list, because an annotation can
-                    # have many tags.
-                    if isinstance(field_value, list):
-                        reversed_order = False
-
-            if reversed_order:
-                lval = cval
-                rval = fval
+                # Test whether a query value appears in a list of field values.
+                return cval in fval
```

Comment:
The fact that the "one_of" operator supports being used in two different ways is weird, but the Hypothesis client currently makes use of this. In this PR I'm maintaining compatibility with the parts of the API used by the client. We can then fix this separately.